x/migration/keeper/msg_server_claim_morse_gateway_test.go

```diff
@@ -1,3 +1,265 @@
+		keepertest.WithBankKeeper(bankKeeper),
+		keepertest.WithGatewayKeeper(gatewayKeeper),
+	}
+
+	k, ctx := keepertest.MigrationKeeper(t, opts...)
+	ctx = ctx.WithBlockHeight(expectedClaimedAtHeight)
+	srv := keeper.NewMsgServerImpl(k)
+
+	morsePrivKey := testmigration.NewMorsePrivateKey(t, 0)
+	morseClaimableAccount := &migrationtypes.MorseClaimableAccount{
+		MorseSrcAddress:  sample.MorseAddressHex(),
+		PublicKey:        morsePrivKey.PubKey().Bytes(),
+		UnstakedBalance:  unstakedBalance,
+		ApplicationStake: zeroUpokt,
+		SupplierStake:    zeroUpokt,
+		// ShannonDestAddress: (intentionally omitted),
+		// ClaimedAtHeight:    (intentionally omitted),
+	}
+
+	accountState := &migrationtypes.MorseAccountState{
+		Accounts: []*migrationtypes.MorseClaimableAccount{
+			morseClaimableAccount,
+		},
+	}
+	accountStateHash, err := accountState.GetHash()
+	require.NoError(t, err)
+
+	// Import the MorseClaimableAccount.
+	_, err = srv.ImportMorseClaimableAccounts(ctx, &migrationtypes.MsgImportMorseClaimableAccounts{
+		Authority:             authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		MorseAccountState:     *accountState,
+		MorseAccountStateHash: accountStateHash,
+	})
+	require.NoError(t, err)
+
+	// Claim the MorseClaimableAccount.
+	morseGatewayStake := gatewaytypes.DefaultMinStake
+	msgClaim, err := migrationtypes.NewMsgClaimMorseGateway(
+		shannonDestAddr,
+		morseClaimableAccount.GetMorseSrcAddress(),
+		morsePrivKey,
+		morseGatewayStake,
+	)
+	require.NoError(t, err)
+
+	msgClaimRes, err := srv.ClaimMorseGateway(ctx, msgClaim)
+	require.NoError(t, err)
+
+	// Construct and assert the expected response.
+	expectedRes := &migrationtypes.MsgClaimMorseGatewayResponse{
+		MorseSrcAddress:     msgClaim.MorseSrcAddress,
+		ClaimedGatewayStake: morseGatewayStake,
+		ClaimedBalance: expectedClaimedUnstakedTokens.
+			Add(morseClaimableAccount.GetSupplierStake()),
+		ClaimedAtHeight: expectedClaimedAtHeight,
+	}
+	require.Equal(t, expectedRes, msgClaimRes)
+
+	// Assert that the persisted MorseClaimableAccount is updated.
+	expectedMorseAccount := morseClaimableAccount
+	expectedMorseAccount.ShannonDestAddress = shannonDestAddr
+	expectedMorseAccount.ClaimedAtHeight = ctx.BlockHeight()
+	foundMorseAccount, found := k.GetMorseClaimableAccount(ctx, msgClaim.MorseSrcAddress)
+	require.True(t, found)
+	require.Equal(t, *expectedMorseAccount, foundMorseAccount)
+
+	// Assert that an event is emitted for each claim.
+	expectedEvent := &migrationtypes.EventMorseGatewayClaimed{
+		ShannonDestAddress:  msgClaim.ShannonDestAddress,
+		MorseSrcAddress:     msgClaim.MorseSrcAddress,
+		ClaimedBalance:      expectedClaimedUnstakedTokens,
+		ClaimedGatewayStake: gatewayStakeToClaim,
+		ClaimedAtHeight:     ctx.BlockHeight(),
+	}
+	claimEvents := events.FilterEvents[*migrationtypes.EventMorseGatewayClaimed](t, ctx.EventManager().Events())
+	require.Equal(t, 1, len(claimEvents))
+	require.Equal(t, expectedEvent, claimEvents[0])
```

Comment:
1. Missing assertion for the staked gateway.
2. Missing existing gateway scenario(s).